### ISSUE TITLE
TransmogrifAI on Apache Zeppelin

### ISSUE THREAD

**Issue by razou** [salesforce/TransmogrifAI#239]
I'm using Apache Zeppelin 0.8.1 and transmogrifai:0.5.1

`z.load("com.salesforce.transmogrifai:transmogrifai-core_2.11:0.5.1")
`
```scala
import com.salesforce.op._
import com.salesforce.op.readers.DataReaders
import com.salesforce.op.features.types._
import com.salesforce.op.features.FeatureBuilder
import com.salesforce.op.stages.impl.classification.BinaryClassificationModelSelector
import com.salesforce.op.stages.impl.classification.BinaryClassificationModelsToTry._ 

import spark.implicits._

case class Passenger
  (
    id: Long,
    survived: Double,
    pClass: Option[Long],
    name: Option[String],
    sex: Option[String],
    age: Option[Double],
    sibSp: Option[Long],
    parCh: Option[Long],
    ticket: Option[String],
    fare: Option[Double],
    cabin: Option[String],
    embarked: Option[String]
  )


val pathToData = "/data/TitanicPassengersTrainData.csv"

val passengersDF = DataReaders.Simple.csvCase[Passenger](path = Some(pathToData)).readDataset().toDF()
passengersDF.count

// Automated feature engineering
val (survived, features) = FeatureBuilder.fromDataFrame[RealNN](passengersDF, response = "survived")
val featureVector = features.transmogrify()

// Automated feature selection
val checkedFeatures = survived.sanityCheck(featureVector, checkSample = 1.0, removeBadFeatures = true)

// Automated model selection
val prediction = BinaryClassificationModelSelector
  .withCrossValidation(seed = 324, numFolds = 10, modelTypesToUse = Seq(OpLogisticRegression,  OpGBTClassifier))
  .setInput(survived, checkedFeatures).getOutput()

val model = new OpWorkflow().setInputDataset(passengersDF).setResultFeatures(prediction).train()

println("Model summary:\n" + model.summaryPretty())
```
This code produces the following error
```
java.lang.AbstractMethodError
  at org.apache.spark.ml.tree.RandomForestParams$class.$init$(treeParams.scala:421)
  at com.salesforce.op.stages.impl.classification.OpRandomForestClassifier.<init>(OpRandomForestClassifier.scala:59)
  at com.salesforce.op.stages.impl.classification.BinaryClassificationModelSelector$Defaults$.modelsAndParams(BinaryClassificationModelSelector.scala:78)
  at com.salesforce.op.stages.impl.selector.ModelSelectorFactory$class.selector(ModelSelectorFactory.scala:87)
  at com.salesforce.op.stages.impl.classification.BinaryClassificationModelSelector$.selector(BinaryClassificationModelSelector.scala:47)
  at com.salesforce.op.stages.impl.classification.BinaryClassificationModelSelector$.withCrossValidation(BinaryClassificationModelSelector.scala:172)
  ... 41 elided
```

Do you have an idea of what may causes this error (trying to use an abstract method) or how can we make it works on apache zeppelin  step by step ?
Thanks

**Comment by tovbinm** [comment]
@razou make sure to also import Apache Spark 2.3.x

**Comment by razou** [comment]
I was using zeppelin in [docker](https://hub.docker.com/r/apache/zeppelin) and it has spark 2.2.1 version. Does this library (transmogrifAI) only works with spark 2.3.x ?

**Comment by tovbinm** [comment]
The latest 0.5.x version works with Spark 2.3.x only, but you can try an older version. See version compatibility matrix - https://github.com/salesforce/TransmogrifAI/blob/master/README.md#adding-transmogrifai-into-your-project

**Comment by razou** [comment]
Thanks @tovbinm , problem solved wen using spark 2.3.3 and TransmogrifAI 0.5.1
